pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -575,24 +609,27 @@ impl DeltaLayerWriterInner {
         };
 
         // Writes summary at the first block (offset 0).
-        let buf = summary.ser_into_page()?;
+        let buf = summary
+            .ser_into_page()
+            .map_err(|e| DeltaLayerWriterError::Other(anyhow::Error::new(e)))?;
         let (_buf, res) = file.write_all_at(buf.slice_len(), 0, ctx).await;
-        res?;
+        res.map_err(|e| DeltaLayerWriterError::Other(anyhow::Error::new(e)))?;
 
         let metadata = file
             .metadata()
             .await
-            .context("get file metadata to determine size")?;
+            .map_err(|e| DeltaLayerWriterError::Other(anyhow::Error::new(e)))?;
```

Comment:
Do it like this. Also in all the other places where you currently use this pattern.
```suggestion
            .map_err(anyhow::Error::new)
            .map_err(DeltaLayerWriterError::Other)?;
```